deepfence_agent/tools/apache/scope/probe/docker/container.go

```diff
@@ -397,8 +397,8 @@ func (c *container) GetParent() *report.Parent {
 }
 
 func (c *container) GetNode() *report.TopologyNode {
-	c.RLock()
-	defer c.RUnlock()
+	c.Lock()
```

Comment:
Do we really need to lock and unlock the complete function ? Is there a way where we can lock and unlock *only* when we require it ?